onnxruntime/core/providers/xnnpack/nn/softmax.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/xnnpack/nn/softmax.h"
+
+#include "core/framework/op_kernel.h"
+#include "core/providers/cpu/math/softmax_shared.h"
+#include "core/optimizer/initializer.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+std::pair<const onnx::TensorProto*, const onnx::TensorProto*> GetQuantizationZeroPointAndScale(const GraphViewer& graphview,
+                                                                                               const NodeUnitIODef& io_def);
+namespace {
+static bool IsQuantSoftmaxSupported(const NodeUnit& node_unit, const GraphViewer& graph) {
+  bool supported = false;
+  do {
+    TensorQuantType x_input_type, output_type;
+    x_input_type = GetTensorQuantType(node_unit, 0, false, graph);
+    output_type = GetTensorQuantType(node_unit, 0, true, graph);
+    if (x_input_type != TensorTypeUint8 ||
+        output_type != TensorTypeUint8) {
+      break;
+    }
+    // to verify softmax output scale and zp is 1/256 and 0, otherwise xnnpack ep has to do extra requantization
+    // idealy, QlinearSoftmax or QDQSoftmax will keep this output scale and zp, but we have to handle some 
+    // qdq models converted from other framework
+    auto [scale_tensor, zero_tensor] = GetQuantizationZeroPointAndScale(graph, node_unit.Outputs()[0]);
+    Initializer q_scale(*scale_tensor, node_unit.ModelPath());
```

Comment:
I tried to avoid using Initializer in the xnnpack code as it's a heavy class and not currently available in a minimal build.

Can fix in a separate PR when we need to make xnnpack available in a minimal build. Need something lightweight like the code in FuseActivation. We don't need to support external data for these small values, but would need to add logic to read the required types from the correct field if the data is not stored in raw_data.

```
            const auto& value = *graph.GetConstantInitializer(arg.Name(), true);
            // these should never be in external data as it makes no sense to put scalars there.
            ORT_ENFORCE(utils::HasExternalData(value) == false,
                        "External data is not supported for the scalar min/max Clip values");

            value_to_set = utils::HasRawData(value)
                               ? *reinterpret_cast<const float*>(value.raw_data().data())
                               : value.float_data()[0];
```

Reply:
Sounds we need a light-version of class `Initializer `. we need it in multiple locations.

I will do it in a seperate PR since this one is huge.